docs/build/tutorials/how-to/deploy-contract.md

```diff
@@ -0,0 +1,480 @@
+
+```javascript
+import "@matterlabs/hardhat-zksync-solc";
+import "@matterlabs/hardhat-zksync-deploy";
+import { HardhatUserConfig } from "hardhat/config";
+
+const config: HardhatUserConfig = {
+  zksolc: {
+    // By not specifying any options, we are using the default settings of zksolc.
+  },
+  solidity: {
+    version: "0.8.17",
+  },
+  defaultNetwork: "zkTestnet",
+  networks: {
+    zkTestnet: {
+      url: "https://sepolia.era.zksync.dev", // The RPC URL of zkSync Era network.
+      ethNetwork: "https://sepolia.infura.io/v3/<API_KEY>", // The Ethereum Web3 RPC URL.
+      zksync: true, // Flag that targets zkSync Era.
+    },
+  },
+};
+```
+
+In the configuration above we have some important fields such as:
+
+- `zksolc` - enabling adjustment and customization of the zksolc compiler.
+- `solidity` - enabling adjustment and customization of the solc compiler.
+- `defaultNetwork` - specifying which network is default to use when running hardhat tasks.
+- `networks` - define which networks we can use in our development. Only one network is used at the time.
+  - `url` is a field containing the URL of the zkSync Era node in case of the zkSync Era network (with zksync flag set to true), or the URL of the Ethereum node. This field is required for all zkSync Era and Ethereum networks.
+  - `ethNetwork` is a field with the URL of the Ethereum node. You can also provide network name (e.g. sepolia) as the value of this field. In this case, the plugin will either use the URL of the appropriate Ethereum network configuration (from the networks section), or the default
+  - `zksync` is a flag that indicates if the network is zkSync Era. This field needs to be set to true for all zkSync Era networks; it is false by default
+
+Find more details about available configuration options in the [official documentation.](../../tooling/hardhat/hardhat-zksync-solc.md#configuration)
+
+Execute the following command in your terminal to run the compilation:
+
+```bash
+yarn hardhat compile
+```
+
+After successful compilation, you should see the output similar to this:
+
+```log
+Compiling contracts for zkSync Era with zksolc v1.3.22 and solc v0.8.17
+Compiling 1 Solidity file
+Successfully compiled 1 Solidity file
+Done in 0.69s.
+```
+
+In the root of your project you will see two new folders that represent zksolc compilation result:
+
+- `artifacts-zk`
+- `cache-zk`
+
+Here are the steps to create deploy script with **hardhat-zksync-deploy** plugin.
+
+1. Navigate to your project's root directory.
+2. Create a new folder named **deploy**.
+3. Inside the **deploy** folder, create a file named **deploy-simple-storage.ts**.
+
+Now we should add some code to the new **deploy-simple-storage.ts** file:
+
+```javascript
+import { Wallet } from "zksync-ethers";
+import * as ethers from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
+
+const PRIVATE_KEY = "YOUR_PRIVATE_KEY_HERE";
```

Comment:
We should not recommend users place the Private Key directly in their script, but instead pull this value from an environment variables and `.env` file